latin_word_syllabification.py

```diff
@@ -0,0 +1,340 @@
+    if they serve as consonants or semivowels. This allows the syllabification
+    algorithm to skip them when finding vowels. For details on semivowels
+    and "u"/"v", see README.
+
+    Where "i" or "y" are serving as semivowels, they are replaced with "j".
+    Where "u" is serving as a semivowel, it is replaced with "w".
+    Where "u" is serving as a consonant, it is replaced with "v".
+
+    word [str]: word to replace characters in
+
+    returns [str]: word with characters replaced
+    """
+    word_w_repl = ""
+    # Handle first character in the word
+    first_let = word[0]
+    try:
+        if (first_let in "y") and (word[1] in "aeiou"):
+            word_w_repl += "j"
+        elif (first_let == "i") and (
+            (word[1] in "aeou") or (word[1] == "h" and word[2] in "aeo")
+        ):
+            word_w_repl += "j"
+        elif (first_let == "u") and (word[1] in "aeiouy"):
+            word_w_repl += "v"
+        else:
+            word_w_repl += first_let
+    except IndexError:
+        # IndexError will only occur on "ih"
+        return word
+    word = word[1:]
+    # Handle remaining characters
+    while word != "":
+        char = word[0]
+        if char not in "iyu" or len(word) == 1:
+            word_w_repl += char
+        elif char in "iy":
+            if ((word_w_repl[-1] in "aeou") and (word[1] in "aeou")) or (
+                (word_w_repl[-2:] not in _CONSONANT_GROUPS)
+                and (word_w_repl[-1] == "h")
+                and (word[1] in "eou")
+            ):
+                word_w_repl += "j"
+            else:
+                word_w_repl += "i"
+        elif char == "u":
+            if word_w_repl[-1] in "qg" and word[1] in "aeiouy":
+                word_w_repl += "w"
+            elif word_w_repl[-1] in "aeiouy" and word[1] in "aeiouy":
+                word_w_repl += "v"
+            else:
+                word_w_repl += "u"
+        word = word[1:]
+    return word_w_repl
+
+
+def _get_vowel_positions(word: str) -> "list[int]":
+    """
+    Gets the positions of vowels in a word.
+
+    Vowels are "a", "e", "i", "o", "u", "y" and long "i" (written "j"),
+    except in such cases where "i"/"j" and "u" are semi-vowels or the written
+    "u" is pronounced "v." We temporarily replace these characters in those cases
+    with other consonants when finding vowel positions.
+
+    word [str]: word to get vowel groups from
+
+    returns [list[int]]: list of string indices of vowels in word
+    """
+    # Replace long "i" ("j") with "i" for the purposes of finding vowels.
+    # Note: places where a semi-vowel was transcribed "j" will be handled
+    # by the _replace_semivowels function.
+    word = word.replace("j", "i")
+    # Replace semi-vowels for the purposes of finding vowels.
+    word = _replace_semivowels_and_v(word)
+    logging.debug("Semivowels, long i's, v's found: %s", word)
+    vowel_positions = []
+    word_indexer = 0
+    while word_indexer < len(word):
```

Comment:
Strings are iterable. 

`for char in word:` will do the same here, and you don't need to worry about keeping track of a counting variable since it will automatically stop when it gets to the end.

```
for idx, char in enumerate(word):
        if char in "aeiouy":
            vowel_positions.append(idx)
```

should do exactly the same as this `while` loop.

Reply:
If you wanted to get really fancy, you could also do:

`vowel_postitions = [idx for idx, char in enumerate(word) if char in "aeiouy"]`